### PR TITLE
r/storagegateway_gateway - add missing `smb_active_directory_settings` arguments + validations.

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_storagegateway_gateway - add `timeout_in_seconds`, `organizational_unit`, `domain_controllers` arguments for `smb_active_directory_settings` block.
resource_aws_storagegateway_gateway - add `smb_active_directory_settings. active_directory_status`, `ec2_instance_id`, `endpoint_type`, `host_environment`, and `gateway_network_interface` attributes.
resource_aws_storagegateway_gateway - add plan time validations for `smb_guest_password`, `smb_active_directory_settings. username`, `smb_active_directory_settings. password`, `smb_active_directory_settings. domain_name`, `gateway_timezone`, and `gateway_name`.
resource_aws_storagegateway_gateway - add support for `medium_changer_type`  value `medium_changer_type`.
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSStorageGatewayGateway_'
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Cached (265.38s)
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthAll (457.53s)
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthUpload (457.54s)
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthDownload (468.90s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayTimezone (360.21s)
--- PASS: TestAccAWSStorageGatewayGateway_disappears (278.12s)
--- PASS: TestAccAWSStorageGatewayGateway_SMBSecurityStrategy (316.96s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbGuestPassword (358.11s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint (360.93s)
--- PASS: TestAccAWSStorageGatewayGateway_tags (421.61s)
--- PASS: TestAccAWSStorageGatewayGateway_CloudWatchLogs (312.24s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout (858.86s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings (871.12s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayName (369.13s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Stored (243.59s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Vtl (271.97s)
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_FileS3 (312.99s)
```

### CODE REVIEW

File: aws/resource_aws_storagegateway_gateway.go [hunk 1]
```diff
@@ -264,25 +335,21 @@ func resourceAwsStorageGatewayGatewayCreate(d *schema.ResourceData, meta interfa
 
 	d.SetId(aws.StringValue(output.GatewayARN))
 
-	if _, err := WaitForStorageGatewayGatewayConnected(conn, d.Id(), d.Timeout(schema.TimeoutCreate)); err != nil {
-		return fmt.Errorf("error waiting for Storage Gateway Gateway activation: %w", err)
+	if _, err = waiter.StorageGatewayGatewayConnected(conn, d.Id(), d.Timeout(schema.TimeoutCreate)); err != nil {
+		return fmt.Errorf("error waiting for waiting for Storage Gateway Gateway (%q) to be Connected: %w", d.Id(), err)
```
Comment (ewbankkit) [inline aws/resource_aws_storagegateway_gateway.go:339]:
```suggestion
		return fmt.Errorf("error waiting for Storage Gateway Gateway (%q) to be Connected: %w", d.Id(), err)
```

File: aws/resource_aws_storagegateway_gateway.go [hunk 2]
```diff
@@ -264,25 +335,21 @@ func resourceAwsStorageGatewayGatewayCreate(d *schema.ResourceData, meta interfa
...
-		log.Printf("[DEBUG] Storage Gateway Gateway %q joining Active Directory domain: %s", d.Id(), m["domain_name"].(string))
+		input := expandStorageGatewayGatewayDomain(v.([]interface{}), d.Id())
+		log.Printf("[DEBUG] Storage Gateway Gateway %q joining Active Directory domain: %s", d.Id(), aws.StringValue(input.DomainName))
 		_, err := conn.JoinDomain(input)
 		if err != nil {
 			return fmt.Errorf("error joining Active Directory domain: %w", err)
 		}
+
+		if _, err = waiter.StorageGatewayGatewayJoinDomainJoined(conn, d.Id()); err != nil {
+			return fmt.Errorf("error waiting for waiting for Storage Gateway Gateway (%q) to be Join domain (%s): %w", d.Id(), aws.StringValue(input.DomainName), err)
```
Comment (ewbankkit) [inline aws/resource_aws_storagegateway_gateway.go:351]:
```suggestion
			return fmt.Errorf("error waiting for Storage Gateway Gateway (%q) to join domain (%s): %w", d.Id(), aws.StringValue(input.DomainName), err)
```

File: aws/resource_aws_storagegateway_gateway.go [hunk 3]
```diff
@@ -493,21 +577,16 @@ func resourceAwsStorageGatewayGatewayUpdate(d *schema.ResourceData, meta interfa
...
-		log.Printf("[DEBUG] Storage Gateway Gateway %q joining Active Directory domain: %s", d.Id(), m["domain_name"].(string))
+		input := expandStorageGatewayGatewayDomain(d.Get("smb_active_directory_settings").([]interface{}), d.Id())
+		log.Printf("[DEBUG] Storage Gateway Gateway %q joining Active Directory domain: %s", d.Id(), aws.StringValue(input.DomainName))
 		_, err := conn.JoinDomain(input)
 		if err != nil {
 			return fmt.Errorf("error joining Active Directory domain: %w", err)
 		}
+
+		if _, err = waiter.StorageGatewayGatewayJoinDomainJoined(conn, d.Id()); err != nil {
+			return fmt.Errorf("error waiting for waiting for Storage Gateway Gateway (%q) to be Join domain (%s): %w", d.Id(), aws.StringValue(input.DomainName), err)
```
Comment (ewbankkit) [inline aws/resource_aws_storagegateway_gateway.go:588]:
```suggestion
			return fmt.Errorf("error waiting for Storage Gateway Gateway (%q) to be Join domain (%s): %w", d.Id(), aws.StringValue(input.DomainName), err)
```

**Review by ewbankkit — APPROVED** [review]
LGTM.

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayGateway_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSStorageGatewayGateway_ -timeout 120m
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== RUN   TestAccAWSStorageGatewayGateway_tags
=== PAUSE TestAccAWSStorageGatewayGateway_tags
=== RUN   TestAccAWSStorageGatewayGateway_GatewayName
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayName
=== RUN   TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== PAUSE TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== RUN   TestAccAWSStorageGatewayGateway_GatewayTimezone
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayTimezone
=== RUN   TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== RUN   TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== PAUSE TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== RUN   TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== PAUSE TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== RUN   TestAccAWSStorageGatewayGateway_disappears
=== PAUSE TestAccAWSStorageGatewayGateway_disappears
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthUpload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthUpload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthDownload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthDownload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthAll
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthAll
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Cached (235.86s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
--- PASS: TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint (346.22s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayTimezone
--- PASS: TestAccAWSStorageGatewayGateway_GatewayTimezone (240.02s)
=== CONT  TestAccAWSStorageGatewayGateway_CloudWatchLogs
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings (906.61s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayName
--- PASS: TestAccAWSStorageGatewayGateway_CloudWatchLogs (200.75s)
=== CONT  TestAccAWSStorageGatewayGateway_tags
--- PASS: TestAccAWSStorageGatewayGateway_GatewayName (274.00s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Vtl
--- PASS: TestAccAWSStorageGatewayGateway_tags (278.14s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Stored
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Vtl (235.54s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_FileS3
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Stored (214.21s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthUpload
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_FileS3 (234.89s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthAll
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthUpload (268.20s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthDownload
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthAll (292.91s)
=== CONT  TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== CONT  TestAccAWSStorageGatewayGateway_disappears
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthDownload (305.92s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbGuestPassword
--- PASS: TestAccAWSStorageGatewayGateway_SMBSecurityStrategy (233.29s)
--- PASS: TestAccAWSStorageGatewayGateway_disappears (219.86s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
--- PASS: TestAccAWSStorageGatewayGateway_SmbGuestPassword (220.10s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout (811.74s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	3120.961s
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @DrFaust92 

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayGateway_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSStorageGatewayGateway_ -timeout 120m
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== RUN   TestAccAWSStorageGatewayGateway_tags
=== PAUSE TestAccAWSStorageGatewayGateway_tags
=== RUN   TestAccAWSStorageGatewayGateway_GatewayName
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayName
=== RUN   TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== PAUSE TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== RUN   TestAccAWSStorageGatewayGateway_GatewayTimezone
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayTimezone
=== RUN   TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== RUN   TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== PAUSE TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== RUN   TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== PAUSE TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== RUN   TestAccAWSStorageGatewayGateway_disappears
=== PAUSE TestAccAWSStorageGatewayGateway_disappears
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthUpload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthUpload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthDownload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthDownload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthAll
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthAll
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Cached (208.89s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthAll
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthAll (273.32s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthDownload
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings (657.51s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthUpload
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthDownload (217.16s)
=== CONT  TestAccAWSStorageGatewayGateway_disappears
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthUpload (231.02s)
=== CONT  TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
--- PASS: TestAccAWSStorageGatewayGateway_disappears (202.49s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbGuestPassword
--- PASS: TestAccAWSStorageGatewayGateway_SMBSecurityStrategy (203.25s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
--- PASS: TestAccAWSStorageGatewayGateway_SmbGuestPassword (250.05s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayName
--- PASS: TestAccAWSStorageGatewayGateway_GatewayName (204.87s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
--- PASS: TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint (279.71s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayTimezone
--- PASS: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout (773.72s)
=== CONT  TestAccAWSStorageGatewayGateway_CloudWatchLogs
--- PASS: TestAccAWSStorageGatewayGateway_GatewayTimezone (236.27s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Vtl
--- PASS: TestAccAWSStorageGatewayGateway_CloudWatchLogs (203.90s)
=== CONT  TestAccAWSStorageGatewayGateway_tags
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Vtl (177.79s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Stored
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Stored (214.65s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_FileS3
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_FileS3 (197.38s) 
--- PASS: TestAccAWSStorageGatewayGateway_tags (257.25s)

```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

Note that failures here are a known issue and are tracked here: #16172

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSStorageGatewayGateway_' ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 2 -run=TestAccAWSStorageGatewayGateway_ -timeout 120m
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_FileS3
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Stored
=== RUN   TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayType_Vtl
=== RUN   TestAccAWSStorageGatewayGateway_tags
=== PAUSE TestAccAWSStorageGatewayGateway_tags
=== RUN   TestAccAWSStorageGatewayGateway_GatewayName
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayName
=== RUN   TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== PAUSE TestAccAWSStorageGatewayGateway_CloudWatchLogs
=== RUN   TestAccAWSStorageGatewayGateway_GatewayTimezone
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayTimezone
=== RUN   TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== PAUSE TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
=== RUN   TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== PAUSE TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
=== RUN   TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== PAUSE TestAccAWSStorageGatewayGateway_SmbGuestPassword
=== RUN   TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== PAUSE TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
=== RUN   TestAccAWSStorageGatewayGateway_disappears
=== PAUSE TestAccAWSStorageGatewayGateway_disappears
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthUpload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthUpload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthDownload
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthDownload
=== RUN   TestAccAWSStorageGatewayGateway_bandwidthAll
=== PAUSE TestAccAWSStorageGatewayGateway_bandwidthAll
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Cached
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings
    resource_aws_storagegateway_gateway_test.go:400: Step 1/2 error: Error running apply: 2020/12/01 13:34:47 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0

        Error: ClientException: Simple AD directory creation is currently not supported in this region. : RequestId: 12b47249-efe8-4467-b8f7-c82d971edbb2 : RequestId: 12b47249-efe8-4467-b8f7-c82d971edbb2
        {
          RespMetadata: {
            StatusCode: 400,
            RequestID: "12b47249-efe8-4467-b8f7-c82d971edbb2"
          },
          Message_: "Simple AD directory creation is currently not supported in this region. : RequestId: 12b47249-efe8-4467-b8f7-c82d971edbb2 : RequestId: 12b47249-efe8-4467-b8f7-c82d971edbb2",
          RequestId: "12b47249-efe8-4467-b8f7-c82d971edbb2"
        }


--- FAIL: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings (19.86s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayName
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Cached (220.26s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint
--- PASS: TestAccAWSStorageGatewayGateway_GatewayName (239.93s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayTimezone
--- PASS: TestAccAWSStorageGatewayGateway_GatewayTimezone (188.78s)
=== CONT  TestAccAWSStorageGatewayGateway_CloudWatchLogs
--- PASS: TestAccAWSStorageGatewayGateway_GatewayVpcEndpoint (251.76s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthUpload
--- PASS: TestAccAWSStorageGatewayGateway_CloudWatchLogs (169.69s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthAll
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthUpload (222.56s)
=== CONT  TestAccAWSStorageGatewayGateway_bandwidthDownload
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthAll (253.21s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Vtl
--- PASS: TestAccAWSStorageGatewayGateway_bandwidthDownload (241.13s)
=== CONT  TestAccAWSStorageGatewayGateway_tags
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Vtl (202.43s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_Stored
--- PASS: TestAccAWSStorageGatewayGateway_tags (260.26s)
=== CONT  TestAccAWSStorageGatewayGateway_SMBSecurityStrategy
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_Stored (211.64s)
=== CONT  TestAccAWSStorageGatewayGateway_disappears
--- PASS: TestAccAWSStorageGatewayGateway_SMBSecurityStrategy (224.80s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbGuestPassword
--- PASS: TestAccAWSStorageGatewayGateway_disappears (188.79s)
=== CONT  TestAccAWSStorageGatewayGateway_GatewayType_FileS3
--- PASS: TestAccAWSStorageGatewayGateway_GatewayType_FileS3 (167.93s)
=== CONT  TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout
    resource_aws_storagegateway_gateway_test.go:430: Step 1/2 error: Error running apply: 2020/12/01 14:02:08 [DEBUG] Using modified User-Agent: Terraform/0.12.29 HashiCorp-terraform-exec/0.10.0

        Error: ClientException: Simple AD directory creation is currently not supported in this region. : RequestId: 57ddbb70-4236-45da-af46-e0ef24f80c02 : RequestId: 57ddbb70-4236-45da-af46-e0ef24f80c02
        {
          RespMetadata: {
            StatusCode: 400,
            RequestID: "57ddbb70-4236-45da-af46-e0ef24f80c02"
          },
          Message_: "Simple AD directory creation is currently not supported in this region. : RequestId: 57ddbb70-4236-45da-af46-e0ef24f80c02 : RequestId: 57ddbb70-4236-45da-af46-e0ef24f80c02",
          RequestId: "57ddbb70-4236-45da-af46-e0ef24f80c02"
        }


--- FAIL: TestAccAWSStorageGatewayGateway_SmbActiveDirectorySettings_timeout (18.73s)
--- PASS: TestAccAWSStorageGatewayGateway_SmbGuestPassword (310.91s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	1733.452s
FAIL
make: *** [testacc] Error 1
```

**Pull request merged**